src/prefect/deployments/base.py

```diff
@@ -446,19 +447,25 @@ def _save_deployment_to_prefect_file(
             ryaml.dump(parsed_prefect_file_contents, f)
 
 
+# Only allow half of the open file limit to be open at once to allow for other
+# actors to open files.
+OPEN_FILE_SEMAPHORE = asyncio.Semaphore(math.floor(get_open_file_limit() * 0.5))
+
+
 async def _find_flow_functions_in_file(filename: str) -> List[Dict]:
     decorator_name = "flow"
     decorator_module = "prefect"
     decorated_functions = []
-    async with await anyio.open_file(filename) as f:
-        try:
-            tree = ast.parse(await f.read())
-        except SyntaxError:
-            if PREFECT_DEBUG_MODE:
-                get_logger().debug(
-                    f"Could not parse {filename} as a Python file. Skipping."
-                )
-            return decorated_functions
+    async with OPEN_FILE_SEMAPHORE:
+        async with await anyio.open_file(filename) as f:
+            try:
+                tree = ast.parse(await f.read())
+            except SyntaxError:
+                if PREFECT_DEBUG_MODE:
+                    get_logger().debug(
+                        f"Could not parse {filename} as a Python file. Skipping."
+                    )
+                return decorated_functions
```

Comment:
Would it make sense to try and catch exceptions here?

Such as `asyncio.BoundedSemaphoreResourceError,` with f"File limit of {limit} reached. Consider adjusting with ulimit

And other `OSError` exceptions like issues with opening the file due to permissions or something

Reply:
I added some generic exception handling so that file permission errors shouldn't crash the CLI. I think it's highly unlikely that this waits indefinitely since files are closed very soon after being parsed. 

Also, I think it's unlikely that users will need to change their open file limit. We're able to scan the entire `prefect` repository (which has almost 600 flows and over 1000 files) in under 5 seconds.

Reply:
👍